### DISCUSSION TITLE
Time & Station ID Announcement (TOTH) in AzuraCast using Text-To-Speech

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#6294] [Show and tell]
Since there seems some demand for top-of-the-hour (TOTH) time and station ID announcements and I use a local Text-to-Speech (PicoTTS) for that, I thought I might as well share my solution with you. I generate an MP3 file every minute that can be played at any time, and speaks out the current day and time as well as a station ID.

I assume you run AzuraCast on a Linux server (Ubuntu 22.04 LTS in my case) and you have SSH access and are allowed to install software (_PicoTTS_ and _sox_) and edit a crontab.

Here is the complete setup: https://blog.syvi.net/tech-snippets/time-und-station-id-announcement-toth-in-azuracast-using-text-to-speech

(The blog is self-hosted and might be down once in a while, just try again later in that case.)

This is how it sounds on a real station:
https://blog.syvi.net/files/matthias/azuracast-toth-real-station-sample.mp3

**Comment by scysys** [comment]
That's a really funny thing, haha.

EDIT

I love it. PicoTTS seems a little bit outdated for nowdays. I searched a little bit for free actual ones and found gTTS.

German: https://voca.ro/1lEY6sHnjgV8
English: https://voca.ro/1mTsjRU5AeME

**Comment by Moonbase59** [comment]
gTTS is exactly what I tried to avoid: Google Text-to-Speech, requiring extra Internet access. It also unfortunately breaks too often, since Google try to keep people from using their Google Translate TTS engine for free. Using gTTS, I have sometimes experienced rather long response times, too.

Nice samples, though.

Too bad SVOX got bought out and commercialized, so no new voices can be had (and the real good old Android voices not being available anymore). I also tried festival and eSpeak, but both are more complicated and the voice quality is rather bad.

**Comment by schulle4u** [comment]
A nice offline solution worth to have an eye on is [Piper](https://github.com/rhasspy/piper). The german voice variants however are currently quite broken in some points.

**Reply by jbillma** [reply]
The great thing about Piper is, if you have the time and a powerful enough computer (pretty much anything with a GPU), you can model your own voice.

**Comment by kryptonmediagroup** [comment]
We use Amazon's "Polly" for this.  It's pretty reliable, costs next to nothing to operate, and sounds decent.
